src/elements/ifc-client-link.ts

```diff
@@ -0,0 +1,30 @@
+import { Client } from '../client';
+const PATH_ATTR = 'path';
+
+/**
+ * A custom element responsible for rendering an anchor element, turning a path
+ * within the current application to a full top-level url.
+ * It is registered as `ifc-client-link` within client.ts
+ */
+
+export default class IfcClientLinkElement extends HTMLElement {
+  private _client: Client;
+
+  /** @internal */
+  constructor(clientInstance: any) {
+    super();
+    this._client = clientInstance;
+
+    const link = document.createElement('a');
+    link.setAttribute('target', '_top');
+    const slot = document.createElement('slot');
+    link.appendChild(slot);
+    const shadowRoot = this.attachShadow({ mode: 'open' });
```

Comment:
Sadly, I think I have to backtrack on the shadow DOM usage here.  It prevents global `a` tag styles from styling these generated links. Not only that, but adding styles to `ifc-client-link` elements doesn't change the internal `a` style unless we explicitly style the `a` tag in this component with `inherit` for things like color and text-decoration, due to the browser's default styles.  But applying `inherit` means that the links come out by default with regular text styles which is also bad. 

Sigh... This is the most awkward part of webcomponents.